examples/jsm/exporters/GLTFExporter.js

```diff
@@ -2962,6 +2969,67 @@ class GLTFMaterialsEmissiveStrengthExtension {
 
 }
 
+/**
+ * GPU Instancing Extension
+ *
+ * Specification: https://github.com/KhronosGroup/glTF/tree/master/extensions/2.0/Vendor/EXT_mesh_gpu_instancing
+ */
+class GLTFMeshGpuInstancing {
+
+	constructor( writer ) {
+
+		this.writer = writer;
+		this.name = 'EXT_mesh_gpu_instancing';
+
+	}
+
+	writeNode( object, nodeDef ) {
+
+		if ( ! object.isInstancedMesh ) return;
+
+		const writer = this.writer;
+
+		const mesh = object;
+
+		const translationAttr = new Float32Array( mesh.count * 3 );
+		const rotationAttr = new Float32Array( mesh.count * 4 );
+		const scaleAttr = new Float32Array( mesh.count * 3 );
+
+		const matrix = new Matrix4();
+		const position = new Vector3();
+		const quaternion = new Quaternion();
+		const scale = new Vector3();
+
+		for ( let i = 0; i < mesh.count; i ++ ) {
+
+			mesh.getMatrixAt( i, matrix );
+			matrix.decompose( position, quaternion, scale );
+
+			position.toArray( translationAttr, i * 3 );
+			quaternion.toArray( rotationAttr, i * 4 );
+			scale.toArray( scaleAttr, i * 3 );
+
+		}
+
+		const attributes = {
+			TRANSLATION: writer.processAccessor( new BufferAttribute( translationAttr, 3 ) ),
+			ROTATION: writer.processAccessor( new BufferAttribute( rotationAttr, 4 ) ),
+			SCALE: writer.processAccessor( new BufferAttribute( scaleAttr, 3 ) ),
+		};
```

Comment:
A comment for potential future optimization. (So no need to do in this PR.)

As commented at https://github.com/takahirox/three-gltf-extensions/blob/00e4233e4fc6c086f13c1c2de80ecc8a3c7eec29/exporters/EXT_mesh_gpu_instancing/EXT_mesh_gpu_instancing_exporter.js#L55
[the attributes are optional](https://github.com/KhronosGroup/glTF/blob/main/extensions/2.0/Vendor/EXT_mesh_gpu_instancing/README.md#extending-nodes-with-instance-attributes) and we may be able to think export an attribute only if it has non-default value. For example, it may not be too rare that all scales are default (1, 1, 1) and in such a case the exported file size can be reducted.

This optimization may be able to be done in gltf post-processing pipeline.

Reply:
We can also merge the PR first and then apply the optimization. In this way, the changes presented in this PR are not blocked.

Reply:
Sure, I can make another PR later.